test/sdn_migration/sdn_migration_test.go

```diff
@@ -0,0 +1,631 @@
+
+	})
+
+	It("rosa cluster is healthy post 4.16.0-rc.0 upgrade", postUpgradeCheck, func(ctx context.Context) {
+		criticalAlerts, _, err := queryPrometheusAlerts(ctx, testRosaCluster.client, fmt.Sprintf("%s/4.16.0-rc.0-prometheus-alerts-pre-upgrade.log", testRosaCluster.reportDir))
+		Expect(err).ShouldNot(HaveOccurred(), "failed to retrieve prometheus alerts")
+		Expect(criticalAlerts).ToNot(BeNumerically(">", 0), "critical alerts are firing pre upgrade")
+		err = osdClusterReadyHealthCheck(ctx, testRosaCluster.client, "post-upgrade", testRosaCluster.reportDir)
+		Expect(err).ShouldNot(HaveOccurred(), "osd-cluster-ready health check job failed post upgrade")
+
+	})
+
+	It("rosa cluster migrated from sdn to ovn successfully", sdnToOvn, func(ctx context.Context) {
+		err := addIntenalTestingAnnotation(ctx, testRosaCluster.client)
+		Expect(err).ShouldNot(HaveOccurred(), "Rosa Cluster failed to patch network")
+		err = patchNetworkConfig(ctx, testRosaCluster.client)
+		Expect(err).ShouldNot(HaveOccurred(), "Rosa Cluster failed to patch network")
+		err = checkMigrationStatus(ctx, testRosaCluster.client)
+		Expect(err).ShouldNot(HaveOccurred(), "Rosa Cluster failed to patch network")
+
+	})
+	It("rosa cluster has no critical alerts firing post sdn to ovn migration", postMigrationCheck, func(ctx context.Context) {
+		criticalAlerts, _, err := queryPrometheusAlerts(ctx, testRosaCluster.client, fmt.Sprintf("%s/prometheus-alerts-pre-upgrade.log", testRosaCluster.reportDir))
+		Expect(err).ShouldNot(HaveOccurred(), "failed to retrieve prometheus alerts")
+		Expect(criticalAlerts).ToNot(BeNumerically(">", 0), "critical alerts are firing pre upgrade")
+		err = osdClusterReadyHealthCheck(ctx, testRosaCluster.client, "post-upgrade", testRosaCluster.reportDir)
+		Expect(err).ShouldNot(HaveOccurred(), "osd-cluster-ready health check job failed post upgrade")
+
+	})
+})
+
+// queryPrometheusAlerts queries prometheus for alerts and provides a count for critical and warning alerts
+func queryPrometheusAlerts(ctx context.Context, client *openshiftclient.Client, logFilename string) (int, int, error) {
+	criticalAlertCount, warningAlertCount := 0, 0
+	alerts := ""
+
+	type metric struct {
+		AlertName  string `json:"alertname"`
+		AlertState string `json:"alertstate"`
+		Condition  string `json:"condition"`
+		Endpoint   string `json:"endpoint"`
+		Name       string `json:"name"`
+		Namespace  string `json:"namespace"`
+		Severity   string `json:"severity"`
+	}
+
+	prometheusClient, _ := prometheusclient.New(ctx, client)
+	vector, err := prometheusClient.InstantQuery(ctx, "ALERTS{alertstate!=\"pending\",alertname!=\"Watchdog\"}")
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to query prometheus: %v", err)
+	}
+
+	for _, model := range vector {
+		metric := metric{}
+
+		metricEncoded, err := json.Marshal(model.Metric)
+		if err != nil {
+			continue
+		}
+
+		err = json.Unmarshal(metricEncoded, &metric)
+		if err != nil {
+			continue
+		}
```

Comment:
I'm not sure I understand why we remarshal the metric into a new struct

Reply:
I'm not entirely sure either; I copied it from https://github.com/openshift/osde2e/blob/main/test/mcscupgrade/mcscupgrade_test.go#L314. Since the goal is to eventually export the metrics to Grafana, I’ve removed it for now.